lib/spack/llnl/util/symlink.py

```diff
@@ -42,21 +42,23 @@ def symlink(real_path, link_path):
 
 
 def islink(path):
-    return os.path.islink(path) or _win32_is_junction(path)
+    return Path(path).is_link() or _win32_is_junction(path)
 
 
 # '_win32' functions based on
 # https://github.com/Erotemic/ubelt/blob/master/ubelt/util_links.py
 def _win32_junction(path, link):
+    path = Path(path)
+    link = Path(link)
     # junctions require absolute paths
-    if not os.path.isabs(link):
-        link = os.path.abspath(link)
+    if not link.is_absolute():
+        link = link.absolute()
```

Comment:
`Path.absolute()` already checks `.is_absolute()`, so we can avoid doing that again here.

```suggestion
    link = link.absolute()
```

Reply:
See https://github.com/python/cpython/blob/0b1d9c44f1f091a499856d81542eeafda25011e1/Lib/pathlib.py#L797-L810:
```python
    def absolute(self):
        """Return an absolute version of this path by prepending the current
        working directory. No normalization or symlink resolution is performed.

        Use resolve() to get the canonical path to a file.
        """
        if self.is_absolute():
            return self
        elif self._drv:
            # There is a CWD on each drive-letter drive.
            cwd = self._flavour.abspath(self._drv)
        else:
            cwd = os.getcwd()
        return self._from_parts([cwd] + self._parts)
```

Reply:
Cool! Thanks for digging into that to verify the behavior.